tests/test_raw_content_spider.py

```diff
@@ -1,11 +1,106 @@
 """ Unit tests for the class RawContentSpider.
 """
+from __future__ import absolute_import
+
+import datetime
 import unittest
 
+import mock
+
+from ketohub.spiders import raw_content_spider
+from scrapy.http import TextResponse, Request
+
+
+class TestSpider(raw_content_spider.RawContentSpider):
+    """ Test class for mocking the RawContentSpider. """
+
+    def __init__(self):
+        self.settings = mock.Mock()
+        self.settings.get.return_value = '/path/to/foo'
+        super(TestSpider, self).__init__()
+
 
 class RawContentSpiderTest(unittest.TestCase):
-    """ Dummy test case. """
+    """ Test case for the raw_content spider. """
+
+    def setUp(self):
+        mock_urllib = mock.patch(
+            'ketohub.spiders.raw_content_spider.urllib.urlretrieve',
+            autospec=True)
+        self.addCleanup(mock_urllib.stop)
+        self.urllib_patch = mock_urllib.start()
+
+        self.mock_start_scrape_time = datetime.datetime(
+            year=2017, day=1, month=1)
+        mock_datetime = mock.patch(
+            'ketohub.spiders.raw_content_spider.datetime')
+        self.addCleanup(mock_datetime.stop)
+        datetime_patch = mock_datetime.start()
+        datetime_patch.utcnow.return_value = self.mock_start_scrape_time
+
+        mock_write_to_file = mock.patch(
+            'ketohub.spiders.raw_content_spider.write_to_file')
+        self.addCleanup(mock_write_to_file.stop)
+        self.write_to_file_patch = mock_write_to_file.start()
+
+        self.spider = TestSpider()
+
+    def build_expected_filepath(self, url):
+        """ Returns the expected filepath for a specific url. """
+        return '{download_root}/{scrape_start_time}/{source_url}'.format(
+            download_root='/path/to/foo',
+            scrape_start_time=self.mock_start_scrape_time.strftime(
+                '%Y%m%d/%H%M%SZ'),
+            source_url=url[12:])
+
+    def test_offline_for_get_image_location_response_from_ketoconnect(self):  #pylint: disable=invalid-name
+        """Tests that parse_recipe wextracts the correct img src for a ketoconnect response."""
+        mock_url = 'https://www.ketoconnect.com/test/'
+        request = Request(mock_url)
+        file_content = "<html><img src='first_image.jpg'><img src='second_image.jpg'></html>"
```

Comment:
We don't need variables for `mock_url`, `request` or `file_content`. We can just declare the values inline in the `TextResponse` constructor and duplicate the URL value in the `assert_called_with`.

Same kind of applies to `response` but I like it because it keeps the [Arrange, Act, Assert](http://wiki.c2.com/?ArrangeActAssert) structure, so let's keep that one as-is.